pkg/vm/compatibility_test.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package vm
+
+import (
+	"testing"
+
+	"github.com/ava-labs/avalanche-cli/internal/mocks"
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/models"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+const (
+	testAvagoVersion         = "v0.4.2"
+	testUnlistedAvagoVersion = "v0.4.3"
+)
+
+var (
+	testSubnetEVMCompat = []byte("{\"rpcChainVMProtocolVersion\": {\"v0.4.2\": 18,\"v0.4.1\": 18,\"v0.4.0\": 17}}")
+	testAvagoCompat     = []byte("{\"19\": [\"v1.9.2\"],\"18\": [\"v1.9.1\"],\"17\": [\"v1.9.0\",\"v1.8.0\"]}")
+	testAvagoCompat2    = []byte("{\"19\": [\"v1.9.2\", \"v1.9.1\"],\"18\": [\"v1.9.0\"]}")
```

Comment:
It might be interesting to have tests with more "tricky" versions, e.g. v1.9.11 vs v.1.9.1 vs 1.9.01 etc.

Reply:
This might be interesting for fuzzing purposes, but I don't think it's really necessary. We're really just testing key value matches. The one part that is interesting is that we're testing some semantic version sorting, which could backfire if some strings aren't semvars, but avalanchego does have protections to make sure that's not the case. So I don't think we really need to do anything else here.

Reply:
Are we sure that v1.9.11 is after v1.9.1 in this PR? v1.10.1 after v1.9.1 ? If the comparison is by string, not necessarily.

Reply:
this was indeed a bug, I added a test and switched the sorting to use `semver` instead of string sorting